tests/filecheck/transforms/copy_to_dma.mlir

```diff
@@ -55,6 +55,95 @@
+//CHECK-NEXT:     %24 = "memref.dim"(%arg0, %23) : (memref<5xi32, strided<[1], offset: ?>, "L0">, index) -> index
+//CHECK-NEXT:     %25 = "arith.constant"() <{"value" = 4 : index}> : () -> index
+//CHECK-NEXT:     %26 = "arith.muli"(%24, %25) : (index, index) -> index
+//CHECK-NEXT:     "func.call"(%8, %15, %26) <{"callee" = @snax_dma_1d_transfer}> : (index, index, index) -> ()
+//CHECK-NEXT:     "func.return"() : () -> ()
+//CHECK-NEXT:   }) : () -> ()
+//CHECK-NEXT:   "func.func"() <{"sym_name" = "snax_dma_1d_transfer", "function_type" = (index, index, index) -> (), "sym_visibility" = "private"}> ({
+//CHECK-NEXT:   }) : () -> ()
+//CHECK-NEXT: }) : () -> ()
+
+// -----
+
+"builtin.module"() ({
+  "func.func"() <{"sym_name" = "transform_copy", "function_type" = (memref<?x?xi32, strided<[?, 1]>, "L0">, memref<?x?xi32, strided<[?, 1]>, "L1">) -> (), "sym_visibility" = "public"}> ({
+  ^0(%arg0 : memref<?x?xi32, strided<[?, 1]>, "L0">, %arg1 : memref<?x?xi32, strided<[?, 1]>, "L1">):
+    "memref.copy"(%arg0, %arg1) : (memref<?x?xi32, strided<[?, 1]>, "L0">, memref<?x?xi32, strided<[?, 1]>, "L1">) -> ()
+    "func.return"() : () -> ()
+  }) : () -> ()
+}) : () -> ()
+
+//CHECK: "builtin.module"() ({
+//CHECK-NEXT:   "func.func"() <{"sym_name" = "transform_copy", "function_type" = (memref<?x?xi32, strided<[?, 1]>, "L0">, memref<?x?xi32, strided<[?, 1]>, "L1">) -> (), "sym_visibility" = "public"}> ({
+//CHECK-NEXT:   ^0(%arg0 : memref<?x?xi32, strided<[?, 1]>, "L0">, %arg1 : memref<?x?xi32, strided<[?, 1]>, "L1">):
+//CHECK-NEXT:     %0 = "memref.extract_aligned_pointer_as_index"(%arg0) : (memref<?x?xi32, strided<[?, 1]>, "L0">) -> index
+//CHECK-NEXT:     %1 = "memref.extract_aligned_pointer_as_index"(%arg1) : (memref<?x?xi32, strided<[?, 1]>, "L1">) -> index
+//CHECK-NEXT:     %2 = "arith.constant"() <{"value" = 0 : index}> : () -> index
+//CHECK-NEXT:     %3 = "memref.dim"(%arg0, %2) : (memref<?x?xi32, strided<[?, 1]>, "L0">, index) -> index
+//CHECK-NEXT:     %4 = "arith.constant"() <{"value" = 1 : index}> : () -> index
+//CHECK-NEXT:     %5 = "memref.dim"(%arg0, %4) : (memref<?x?xi32, strided<[?, 1]>, "L0">, index) -> index
+//CHECK-NEXT:     %6 = "arith.constant"() <{"value" = 1 : index}> : () -> index
+//CHECK-NEXT:     %7 = "arith.divui"(%3, %6) : (index, index) -> index
+//CHECK-NEXT:     %8 = "arith.constant"() <{"value" = 1 : index}> : () -> index
+//CHECK-NEXT:     %9 = "arith.divui"(%5, %8) : (index, index) -> index
+//CHECK-NEXT:     %10 = "arith.constant"() <{"value" = 4 : index}> : () -> index
+//CHECK-NEXT:     %11 = "arith.constant"() <{"value" = 4 : index}> : () -> index
+//CHECK-NEXT:     %12 = "arith.muli"(%9, %10) : (index, index) -> index
+//CHECK-NEXT:     %13 = "arith.constant"() <{"value" = 4 : index}> : () -> index
+//CHECK-NEXT:     %14 = "arith.constant"() <{"value" = 4 : index}> : () -> index
+//CHECK-NEXT:     %15 = "arith.muli"(%9, %13) : (index, index) -> index
```

Comment:
if the algorithm detects that it is contiguous, then we can maybe elide emitting this?
I know this will get dce'd, but maybe we don't even have to insert it in the first place?

Reply:
I'll keep this in mind, but I think it is quite difficult to do this. The code is already complicated enough, and if dce works properly it is not worth it to complicate this further imo

Reply:
Sure! I don't think it's a priority.
Just because you said:
> Although this filecheck is getting a bit out of nad.